ApiEverest/CrudApi/Services/CustomerService.cs

```diff
@@ -0,0 +1,63 @@
+﻿using CustomerApi.Models.Entities;
+
+namespace CustomerApi.Services
+{
+    public class CustomerService : ICustomerService
+    {
+        private readonly List<CustomerEntity> listCustomers = new List<CustomerEntity>();
+
+        public bool Create(CustomerEntity model)
+        {
+            model.Id = listCustomers.LastOrDefault()?.Id + 1 ?? 1;
+            listCustomers.Add(model);   
+            return true;
+        }
+
+        public bool Delete(long id)
+        {
+           int index = listCustomers.FindIndex(customer => customer.Id == id);
+            listCustomers.RemoveAt(index);
+            return true;
+        }
+
+        public IEnumerable<CustomerEntity> GetAll()
+        {
+            return listCustomers;
+        }
+
+        public CustomerEntity? GetById(long id)
+        {
+          return listCustomers.Find(customer => customer.Id == id);
```

Comment:
Aqui usa o FirstOrDefault no lugar do Find e não retorna direto a pesquisa, salva primeiro em uma variável e depois retorna ela, facilita para fazer debug.

Também precisa criar uma exeption para caso venha nulo, dessa forma:

if (customer == null)
                throw new ArgumentNullException($"Customer with id: {id} was not found");